diff --git a/dist/server/index.js b/dist/server/index.js
index d61f6716a7b49354722aa01f45465c1f2e912006..8568d24a144e09b1f23ebbd1adb652e7f97c5c2f 100644
--- a/dist/server/index.js
+++ b/dist/server/index.js
@@ -1,22 +1,20 @@
 import { getDirname, path } from '@vuepress/utils';
 let __dirname = getDirname(import.meta.url);
-export let umamiAnalyticsPlugin = ({ doNotTrack, autoTrack, hostUrl, domains, cache, src, id, }) => app => {
-    let plugin = {
-        name: 'vuepress-plugin-umami-analytics',
-    };
-    if (app.env.isDev) {
-        return plugin;
-    }
+export let umamiAnalyticsPlugin = ({ doNotTrack, autoTrack, hostUrl, domains, cache, src, id, }) => {
     return {
-        ...plugin,
-        define: {
-            __UMAMI_ANALYTICS_DO_NOT_TRACK__: doNotTrack,
-            __UMAMI_ANALYTICS_AUTO_TRACK__: autoTrack,
-            __UMAMI_ANALYTICS_HOST_URL__: hostUrl,
-            __UMAMI_ANALYTICS_DOMAINS__: domains,
-            __UMAMI_ANALYTICS_CACHE__: cache,
-            __UMAMI_ANALYTICS_SRC__: src,
-            __UMAMI_ANALYTICS_ID__: id,
+        name: 'vuepress-plugin-umami-analytics',
+        define: app => {
+            if (!app.env.isDev) {
+                return {
+                    __UMAMI_ANALYTICS_DO_NOT_TRACK__: doNotTrack || false,
+                    __UMAMI_ANALYTICS_AUTO_TRACK__: autoTrack || false,
+                    __UMAMI_ANALYTICS_HOST_URL__: hostUrl || null,
+                    __UMAMI_ANALYTICS_DOMAINS__: domains || [],
+                    __UMAMI_ANALYTICS_CACHE__: cache || false,
+                    __UMAMI_ANALYTICS_SRC__: src,
+                    __UMAMI_ANALYTICS_ID__: id,
+                };
+            }
         },
         clientConfigFile: path.resolve(__dirname, '../client/index.js'),
     };
